packages/nodes-base/nodes/InstagramBasicDisplay/MediaDescription.ts

```diff
@@ -0,0 +1,165 @@
+import {
+	INodeProperties,
+} from 'n8n-workflow';
+
+export const mediaOperations = [
+	{
+		displayName: 'Operation',
+		name: 'operation',
+		type: 'options',
+		displayOptions: {
+			show: {
+				resource: [
+					'media',
+				],
+			},
+		},
+		options: [
+			{
+				name: 'Get All',
+				value: 'getAll',
+				description: 'Get all images, videos and albums or all fields and edges.',
+			},
+		],
+		default: 'getAll',
+		description: 'Operation to perform.',
+	},
+] as INodeProperties[];
+
+export const mediaFields = [
+	{
+		displayName: 'Type',
+		name: 'type',
+		type: 'options',
+		required: true,
+		default: 'userMedia',
+		description: 'The type of media to be returned',
+		displayOptions: {
+			show: {
+				resource: [
+					'media',
+				],
+				operation: [
+					'getAll',
+				],
+			},
+		},
+		options: [
+			{
+				name: 'User media',
+				value: 'userMedia',
+				description: 'A list of images, videos, or albums on a user',
+			},
+			{
+				name: 'Album media',
+				value: 'albumMedia',
+				description: 'A list of images and videos on an album',
+			},
+			{
+				name: 'Media fields and edges',
+				value: 'mediaFieldsAndEdges',
+				description: 'Fields and edges on an image, video, or album',
+			},
+		],
+	},
+	{
+		displayName: 'User ID for media',
+		name: 'userIdForMedia',
+		type: 'string',
+		default: '',
+		description: 'The ID of the user whose media is to be returned',
+		displayOptions: {
+			show: {
+				resource: [
+					'media',
+				],
+				operation: [
+					'getAll',
+				],
+				type: [
+					'userMedia',
+				],
+			},
+		},
+	},
+	{
+		displayName: 'Media ID',
+		name: 'mediaId',
+		type: 'string',
+		default: '',
+		description: 'The ID of the media to be returned',
+		displayOptions: {
+			show: {
+				resource: [
+					'media',
+				],
+				operation: [
+					'getAll',
+				],
+				type: [
+					'albumMedia',
+					'mediaFieldsAndEdges',
+				],
+			},
+		},
+	},
+	{
+		displayName: 'Fields',
+		name: 'fields',
+		type: 'multiOptions',
+		options: [
+			{
+				name: 'Caption',
+				value: 'caption',
+			},
+			{
+				name: 'ID',
+				value: 'id',
+			},
+			{
+				name: 'Media type',
+				value: 'media_type',
```

Comment:
I think it should be camel cased?

Reply:
To be honest, I'm not really sure. I agree that most are camelCased, but I used snake_case for the Hacker News node and that was okay. The advantage of using snake_case is that in this case this value will be used in the request as is, because that's what the API expects, i.e. this way we do away with the need to convert the value. I'll ask Ricardo about this today.

Reply:
Ricardo just confirmed `value` in `options` can be camelCase or snake_case - both are valid.